code/system_paths.py

```diff
@@ -0,0 +1,55 @@
+from talon import Module, Context, app, actions
+from .user_settings import get_list_from_csv
+import os
+
+mod=Module()
+ctx = Context()
+
+mod.list("system_paths", desc="List of system paths")
+
+user_path = os.path.expanduser("~")
+
+def populate_system_paths():
+    default_system_paths = {
+        "user": user_path,
+        "profile": user_path,
+        "desktop": os.path.join(user_path, "Desktop"),
+        "desk": os.path.join(user_path, "Desktop"),
+        "documents": os.path.join(user_path, "Documents"),
+        "docks": os.path.join(user_path, "Documents"),
+        "downloads": os.path.join(user_path, "Downloads"),
+        "music": os.path.join(user_path, "Music"),
+        "pictures": os.path.join(user_path, "Pictures"),
+        "videos": os.path.join(user_path, "Videos"),
+        "talon home": str(actions.path.talon_home()),
+        "talon user": str(actions.path.talon_user())
+    }
+
+    if app.platform == "windows":
+        one_drive_path = os.path.expanduser(os.path.join("~", "OneDrive"))
+
+        # this is probably not the correct way to check for onedrive, quick and dirty
+        if os.path.isdir(os.path.expanduser(os.path.join("~", r"OneDrive\Desktop"))):
+
+            onedrive_paths = {
+                "desktop": os.path.join(one_drive_path, "Desktop"),
+                "documents": os.path.join(one_drive_path, "Documents"),
+                "one drive": one_drive_path,
+                "pictures": os.path.join(one_drive_path, "Pictures"),
+            }
+        
+            default_system_paths.update(onedrive_paths)
+
+    system_paths = get_list_from_csv(
+        "system_paths.csv",
+        headers=("Path", "Spoken"),
+        default=default_system_paths
+    )
+
+    ctx.lists["user.system_paths"] = system_paths
+
+@mod.capture(rule="{user.system_paths}")
+def system_path(m) -> str:
+    return m.system_paths
+
+app.register("ready", populate_system_paths)
```

Comment:
I did that because I got an error if I called `actions.path.talon_home()` and `actions.path.talon_user()` before ready. If you think it's worth it I can't populate the paths at the start and then add those two paths after ready. Also I think I should rename that function to `on_ready`

Reply:
I was thinking that maybe I can get those paths without using those functions. I know how to get the paths for Linux and Mac, not sure for Windows

Reply:
Oh fair enough, no problem with using the ready listener in that case.